### PR TITLE
chore: migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by noahdietz**
* move unit tests
* automate release creation on tag push
* update readme
* update required checks in sync-repo-settings
* remove extraneous release script

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        with:
+          tag_name: ${{ needs.inspect.outputs.version }}
+          release_name: ${{ needs.inspect.outputs.version }
+          draft: false
+          prerelease: false
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
```
Comment (vchudnov-g) [inline .github/workflows/release.yaml:37]: In the RHS, is `upload_url` implicitly defined as part of `create_release`?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            arch: amd64
+          - os: windows
+            arch: amd64
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.13"
+      - uses: actions/checkout@v2
+      # This project does not use these,  but we need them to build the
+      # binaries.
```
Comment (vchudnov-g) [inline .github/workflows/release.yaml:60]: I find this comment confusing. Could you clarify?

File: README.md [hunk 3]
```diff
@@ -173,35 +173,8 @@ If developing the `grpc-fallback-go` project, run tests via `make test`.
...
-  ```sh
-  gcloud auth configure-docker
-  gcloud docker -- push gcr.io/gapic-images/fallback-proxy
-  gcloud docker -- push gcr.io/gapic-images/fallback-proxy:$VERSION
-  ```
-6. (optional) Clean up!
-```sh
-make clean
-```
+Create a new tag with the form `vX.Y.Z` and push it `git push --tags`. GitHub
```
Comment (vchudnov-g) [inline README.md:176]: Consider: "To make a release, create...."

Comment (vchudnov-g) [inline README.md:176]: "push it using"

**Review by vchudnov-g — APPROVED** [review]
Minor questions and suggestions.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (noahdietz) [reply .github/workflows/release.yaml:60]: They are not a compile-time dependency consumed by the code being compiled, rather they are build tools.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (noahdietz) [reply .github/workflows/release.yaml:37]: Yeah, `create_release` is a GitHub 1st party action. https://github.com/actions/create-release#outputs

File: README.md (hunk 3 above)
Comment (noahdietz) [reply README.md]: Done.

Comment (noahdietz) [reply README.md]: Done.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (noahdietz) [reply .github/workflows/release.yaml]: I clarified it a bit more.

**Pull request merged**